<br><br>
# <center>STATISTICAL METHODS IN FINANCE</center>
# <center>HW4</center>
<center>Fan Yang</center>
<center>UNI: fy2232</center>
<center>02/21/2018</center>

# Problem 1

***For each of the following two covariance matrices, perform the principal component decomposition, i.e. calculate its eigenvalues and corresponding eigenvectors. What proportion of the total variance does the first principal component explain in each case?<br>***
>(a)  $~\left(\begin{matrix}
   1 & \rho \\
   \rho & 1 \\
  \end{matrix}\right)$.<br><br>
(b)  $~\left(\begin{matrix}
   1 & \rho & 0\\
   \rho & 1 & 0\\
   0 & 0 & 1
  \end{matrix}\right)$.
  
Hint: You could refer to Homework 1 for reference.

## (a)
Denote $A$ as the matrix.<br>
First caculate the eigen value.
\begin{split}
|A-\lambda I|&=
\left|\begin{matrix}
   1-\lambda & \rho \\
   \rho & 1-\lambda \\
  \end{matrix}\right|\\
&=(1-\lambda)^2-\rho^2\\
&=(1-\lambda+\rho)(1-\lambda-\rho)\\
\text{so, }\lambda_1&=1+\rho,~ ~\lambda_2=1-\rho.\\
\end{split}
Then caculate the eigen vectors.<br>
\begin{split}
A-\lambda_1I&=
\left(\begin{matrix}
   1-\lambda & \rho \\
   \rho & 1-\lambda \\
  \end{matrix}\right)\\
&=
\left(\begin{matrix}
   -\rho & \rho \\
   \rho & -\rho \\
  \end{matrix}\right)\\
  &=\rho
\left(\begin{matrix}
   -1 & 1 \\
   1 & -1 \\
  \end{matrix}\right)\\
    &\longrightarrow\rho
\left(\begin{matrix}
   -1 & 1 \\
   0 &  0\\
  \end{matrix}\right)\\
\end{split}
So the eigen value for $\lambda_1$ is $(1,1)$.
\begin{split}
A-\lambda_2I
&=
\left(\begin{matrix}
   \rho & \rho \\
   \rho & \rho \\
  \end{matrix}\right)\\
  &=\rho
\left(\begin{matrix}
   1 & 1 \\
   1 & 1 \\
  \end{matrix}\right)\\
    &\longrightarrow\rho
\left(\begin{matrix}
   1 & 1 \\
   0 &  0\\
  \end{matrix}\right)\\
\end{split}
So the eigen value for $\lambda_2$ is $(1,-1)$.<br><br>
Therefore, the orthonormal eigen vectors are $(\frac{1}{\sqrt{2}},\frac{1}{\sqrt{2}})$ and $(\frac{1}{\sqrt{2}},-\frac{1}{\sqrt{2}})$.<br><br>
When $\rho>0$, $\lambda_\max=1+\rho$. The first principal component is $Y_1 = \frac{1}{\sqrt{2}}X_1+\frac{1}{\sqrt{2}}X_2$.
$$
\frac{Var(Y_1)}{Var(Y_1)+Var(Y_2)}=\frac{\lambda_1}{\lambda_1+\lambda_2}
=\frac{1+\rho}{2}
$$
When $\rho\leq0$, $\lambda_\max=1-\rho$. The first principal component is $Y_1 = \frac{1}{\sqrt{2}}X_1-\frac{1}{\sqrt{2}}X_2$.
$$
\frac{Var(Y_1)}{Var(Y_1)+Var(Y_2)}=\frac{\lambda_1}{\lambda_1+\lambda_2}
=\frac{1-\rho}{2}
$$

## (b)
Denote $A$ as the matrix.
\begin{split}
|A-\lambda I|&=
\left|\begin{matrix}
   1-\lambda & \rho & 0\\
   \rho & 1-\lambda & 0\\
   0 & 0 & 1-\lambda\\
  \end{matrix}\right|\\
&=(1-\lambda)\left[(1-\lambda)^2-\rho^2\right]\\
&=(1-\lambda)(1-\lambda+\rho)(1-\lambda-\rho)\\
\text{so, }\lambda_1&=1,~ ~\lambda_2=1+\rho,~ ~\lambda_3=1-\rho.\\
\end{split}
Then caculate the eigen vectors.<br>
\begin{split}
A-\lambda_1I
&=
\left(\begin{matrix}
   0 & \rho &0 \\
   \rho & 0 &0\\
      0 & 0 &0\\
  \end{matrix}\right)\\
    &\longrightarrow
\left(\begin{matrix}
   0 & 1 & 0\\
   1 &  0 &0\\
    0 &  0 &0\\
  \end{matrix}\right)\\
\end{split}
So the eigen value for $\lambda_1$ is $(0,0,1)$.<br>
\begin{split}
A-\lambda_2I
&=
\left(\begin{matrix}
   -\rho & \rho &0 \\
   \rho & -\rho &0\\
      0 & 0 &-\rho\\
  \end{matrix}\right)\\
    &\longrightarrow
\left(\begin{matrix}
   -1 & 1 & 0\\
   0 &  0 &-1\\
    0 &  0 &0\\
  \end{matrix}\right)\\
\end{split}
So the eigen value for $\lambda_2$ is $(\frac{1}{\sqrt{2}},\frac{1}{\sqrt{2}},0)$.<br>
\begin{split}
A-\lambda_2I
&=
\left(\begin{matrix}
   \rho & \rho &0 \\
   \rho & \rho &0\\
      0 & 0 &\rho\\
  \end{matrix}\right)\\
    &\longrightarrow
\left(\begin{matrix}
   1 & 1 & 0\\
   0 &  0 &1\\
    0 &  0 &0\\
  \end{matrix}\right)\\
\end{split}
So the eigen value for $\lambda_2$ is $(\frac{1}{\sqrt{2}},-\frac{1}{\sqrt{2}},0)$.<br><br>
Therefore, the orthonormal eigen vectors are $(0,0,1)$ and $(\frac{1}{\sqrt{2}},\frac{1}{\sqrt{2}},0)$ and $(\frac{1}{\sqrt{2}},-\frac{1}{\sqrt{2}},0)$.<br><br>
When $\rho>0$, then $\lambda_\max=1+\rho$
$$
\frac{Var(Y_1)}{Var(Y_1)+Var(Y_2)+Var(Y_3)}=\frac{\lambda_2}{\lambda_1+\lambda_2+\lambda_3}
=\frac{1+\rho}{3}
$$
When $\rho<0$, then $\lambda_\max=1-\rho$
$$
\frac{Var(Y_1)}{Var(Y_1)+Var(Y_2)+Var(Y_3)}=\frac{\lambda_3}{\lambda_1+\lambda_2+\lambda_3}
=\frac{1-\rho}{3}
$$
When $\rho=0$, then $\lambda_\max=1$
$$
\frac{Var(Y_1)}{Var(Y_1)+Var(Y_2)+Var(Y_3)}=\frac{\lambda_\max}{\lambda_1+\lambda_2+\lambda_3}
=\frac{1}{3}
$$

# Problem 2

***The file “FamaFrench_mon_69_98.txt” contains monthly figures (in %) of the market portfolio excess return (Mkt.RF), SMB, HML and the risk-free rate (RF) from Jan 1969 to Dec 1998 (i.e. 360 observations). On the other hand, the data set “CRSPmon” in the “Ecdat” R-package includes monthly returns of General Electric (GE), IBM and Mobil over the same period. You may use the following two lines of codes for access to the “CRSPmon” data.<br><br>***
`library("Ecdat")`<br>
`data(CRSPmon)`
>***(a)*** Perform principal component analysis (PCA) on the monthly returns of GE, IBM and Mobil. Report the following quantities.<br>
>>(i) The proportion of total variance explained by the first principal component. <br>
(ii) The loading vectors (the eigenvectors).<br>
(iii) The principal components for the first two observations.<br>

>***(b)*** Now, for each of following cases, fit the Fama-French three factor model on GE, IBM and Mobil respectively. Comment on any difference between the two cases.<br>
>>(i) Using data in the first 180 months only;<br>
(ii) Using data in the last 180 months only.<br>
Hint: You may refer to “fama-french-3f.pdf” and “FF3.R” for an illustration.

Load the data:

In [90]:
library(Ecdat)
CRSP <- CRSPmon[,-4]
FamaFrench <- read.table("FamaFrench_mon_69_98.txt", header = T)
head(FamaFrench,5)

date   Mkt.RF SMB   HML   RF  
1 196901 -1.20  -0.80  1.57 0.53
2 196902 -5.82  -3.90  0.93 0.46
3 196903  2.59  -0.28 -0.45 0.46
4 196904  1.52  -0.85  0.06 0.53
5 196905  0.02  -0.27  0.74 0.48

## (a)

### (i)

In [81]:
CRSP.pca <- prcomp(CRSP,scale = T)
summary(CRSP.pca)

Importance of components:
                          PC1    PC2    PC3
Standard deviation     1.3103 0.8773 0.7165
Proportion of Variance 0.5723 0.2566 0.1711
Cumulative Proportion  0.5723 0.8289 1.0000

The proportion of variance explained by the first principal component is 57.23%.

### (ii)

In [82]:
CRSP.pca

Standard deviations (1, .., p=3):
[1] 1.3103474 0.8773128 0.7164578

Rotation (n x k) = (3 x 3):
            PC1        PC2        PC3
ge    0.6284832 -0.2093695 -0.7491150
ibm   0.5937757 -0.4929652  0.6359369
mobil 0.5024334  0.8444819  0.1855019

The eigen vectors are $$(0.6284832,0.5937757,0.5024334)$$ and $$(-0.2093695,-0.4929652,0.8444819)$$ and $$(-0.7491150,0.6359369,0.1855019).$$

### (iii)

In [89]:
CRSP.pca$x[1:2,]

PC1       PC2        PC3       
[1,] -1.079746  0.2351180 -0.4131834
[2,] -1.590621 -0.8147668  0.4647787

## (b)

## (i)

In [107]:
library("robust")
data.b1 <- FamaFrench[1:180,]
CRSP.b1 <- CRSPmon[1:180,]
ge = 100*CRSP.b1[,1] - data.b1$RF
ibm = 100*CRSP.b1[,2] - data.b1$RF
mobil = 100*CRSP.b1[,3] - data.b1$RF
stocks=cbind(ge,ibm,mobil)
fit = lm(cbind(ge,ibm,mobil)~Mkt.RF+SMB+HML, data = data.b1)
options(digits=3)
fit


Call:
lm(formula = cbind(ge, ibm, mobil) ~ Mkt.RF + SMB + HML, data = data.b1)

Coefficients:
             ge      ibm     mobil 
(Intercept)   0.322   0.226   0.390
Mkt.RF        1.133   0.807   1.146
SMB          -0.334  -0.172  -0.586
HML           0.107  -0.158   0.315


## (ii)

In [109]:
library("robust")
data.b2 <- FamaFrench[181:360,]
CRSP.b2 <- CRSPmon[181:360,]
ge = 100*CRSP.b2[,1] - data.b2$RF
ibm = 100*CRSP.b2[,2] - data.b2$RF
mobil = 100*CRSP.b2[,3] - data.b2$RF
stocks=cbind(ge,ibm,mobil)
fit = lm(cbind(ge,ibm,mobil)~Mkt.RF+SMB+HML, data = data.b2)
options(digits=3)
fit


Call:
lm(formula = cbind(ge, ibm, mobil) ~ Mkt.RF + SMB + HML, data = data.b2)

Coefficients:
             ge        ibm       mobil   
(Intercept)   0.33936  -0.00054   0.14640
Mkt.RF        1.12173   0.74824   0.86945
SMB          -0.46179  -0.61333  -0.15081
HML          -0.16594  -0.61853   0.50187


-All three stocks possess negative coeffcients on SMB, which means they perform like big stocks.<br><br>
-For the first 180 months, GE and Mobil have positive coecients on HML, suggesting that they behave like value stocks (high book-to-market value).
While for the last 180 months, GE have negative coefficients on HML, which means it turns to low book-to-market value.<br><br>
-IBM always has negative coeffcients on HML( no matter first or last 180 months), so it is more like a growth stock (low book-to-market value).

# Problem 3

***As in the oil drilling example (see “capm-oil-drilling-example.pdf” in folder “Lecture 4”), suppose we have the same market parameters: $\mu_f = 10\%$, $\mu_M = 17\%$ and $\sigma_M = 12\%$. Suppose further that the expected share price of the oil drilling company after one year is also \$1000/share.<br>***
>*** (a)***If the oil drilling company has a beta of 0.5, what is its fair price according to the SML? <br>
*** (b)*** If its beta is 0.7, what should be its fair price then?<br>

## (a).
Suppose that the fair price is $P_0$. According to SML,
$$
\frac{1000-P_0}{P_0}=\mu_f+\beta(\mu_M-\mu_f)\\
\frac{1000-P_0}{P_0}=0.10+0.5\times(0.17-0.10)\\
P_0=\frac{1000}{1+0.10+0.5\times(0.17-0.10)}=881.057\\
$$


## (b).
Suppose that the fair price is $P_0$. According to SML,
$$
\frac{1000-P_0}{P_0}=\mu_f+\beta(\mu_M-\mu_f)\\
\frac{1000-P_0}{P_0}=0.10+0.7\times(0.17-0.10)\\
P_0=\frac{1000}{1+0.10+0.7\times(0.17-0.10)}=870.322\\
$$
